keps/sig-testing/2464-kubetest2-ci-migration/README.md

```diff
@@ -0,0 +1,743 @@
+    - We propose enforcing no further changes to the remaining deprecated tools to incentivize community migration and improvements to kubetest2.
+    - Other community efforts are already underway (e.g. [removing the kubernetes_build scenario](https://github.com/kubernetes/release/issues/1711))
+- Backporting changes to older release branches
+    - In general, no changes should be made to the release branches or their CI and the old jobs will age out.
+- Decoupling from kubekins to a more minimal image
+    - The kubekins image has become a kitchen-sink over the years, used by more jobs than necessary, and containing more dependencies than required for effective kubetest2 usage
+    - Migrating kubekins away from bootstrap as a base image is only feasible once all jobs have migrated away from bootstrap, which is a non-goal of this KEP
+
+
+## Proposal
+
+<!--
+This is where we get down to the specifics of what the proposal actually is.
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation. What is the desired outcome and how do we measure success?.
+The "Design Details" section below is for the real
+nitty-gritty.
+-->
+
+For the development (main/master) branch only, **NOT** the existing release branches:
+
+- Create shadow canary jobs for each [presubmit-kubernetes-blocking](https://testgrid.k8s.io/presubmits-kubernetes-blocking), (release-blocking)[https://testgrid.k8s.io/sig-release-master-blocking] to use kubetest2 as optional, non-blocking
+- Monitor these canary jobs till they are stable
+- Make kubetest2 jobs non-optional, non-blocking
+- Make kubetest2 jobs non-optional, blocking
+
+No changes should be made to the release branches or their CI.
+
+### User Stories (Optional)
+
+<!--
+Detail the things that people will be able to do if this KEP is implemented.
+Include as much detail as possible so that people can understand the "how" of
+the system. The goal here is to make this feel real for users without getting
+bogged down.
+-->
+
+#### Story 1
+
+#### Story 2
+
+### Notes/Constraints/Caveats (Optional)
+
+<!--
+What are the caveats to the proposal?
+What are some important details that didn't come across above?
+Go in to as much detail as necessary here.
+This might be a good place to talk about core concepts and how they relate.
+-->
+
+### Risks and Mitigations
+
+<!--
+What are the risks of this proposal, and how do we mitigate? Think broadly.
+For example, consider both security and how this will impact the larger
+Kubernetes ecosystem.
+
+How will security be reviewed, and by whom?
+
+How will UX be reviewed, and by whom?
+
+Consider including folks who also work outside the SIG or subproject.
+-->
+
+- Stage 1 of  shipping the kubetest2 binary involves keeping HEAD at kubetest2 unbroken (same as kubetest currently). 
```

Comment:
@hasheddan we probably should target a v1.0.0/v1 release of k/release to freeze our API.

Reply:
@saschagrunert would that be required as part of this KEP?

Reply:
I think we can keep this a separate topic in k/release, WDYT?